### PR TITLE
flatpak: Improve setup and fix Vala when Sdk extension is not installed

### PR DESCRIPTION
**Pull request by sonnyp**
Partially fixes https://github.com/sonnyp/Workbench/issues/604 (Vala)

### CODE REVIEW

File: build-aux/re.sonny.Workbench.Devel.json [hunk 1]
```diff
@@ -3,34 +3,37 @@
...
-    "append-path": "/usr/lib/sdk/rust-stable/bin:/usr/lib/sdk/vala/bin",
+    "no-debuginfo": true,
+    "append-path": "/usr/lib/sdk/vala/bin:/usr/lib/sdk/rust-stable/bin:/usr/lib/sdk/llvm16/bin",
     "append-ld-library-path": "/usr/lib/sdk/vala/lib",
-    "test-args": ["--socket=x11", "--share=network"]
+    "env": {
+      "CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER": "clang",
+      "CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUSTFLAGS": "-C link-arg=-fuse-ld=/usr/lib/sdk/rust-stable/bin/mold",
+      "CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER": "clang",
+      "CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_RUSTFLAGS": "-C link-arg=-fuse-ld=/usr/lib/sdk/rust-stable/bin/mold"
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply build-aux/re.sonny.Workbench.Devel.json:21]: We care for mold only for interactive use. So it is not important to have here

File: src/workbench [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+export WEBKIT_DISABLE_DMABUF_RENDERER=1
+# export G_MESSAGES_DEBUG=@app_id@
+
+export CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=clang
+export CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUSTFLAGS="-C link-arg=-fuse-ld=/app/bin/mold"
+export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=clang
+export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_RUSTFLAGS="-C link-arg=-fuse-ld=/app/bin/mold"
+
+# Required when org.freedesktop.Sdk.Extension.vala is not installed
+export LD_LIBRARY_PATH=/app/lib/vala-0.56/:$LD_LIBRARY_PATH
```
[the remark this answers is not in the document]
Comment (lw64) [reply src/workbench:12]: This looks correct.

File: build-aux/re.sonny.Workbench.Devel.json [hunk 3]
```diff
@@ -58,16 +57,14 @@
...
       ],
       "post-install": [
-        "rm -rf /app/share/doc/appstream",
-        "cp -a /usr/share/gtk-doc/html/. /app/share/doc/",
-        "rm -rf /app/share/doc/appstream",
-        "cp -a /usr/share/doc/. /app/share/doc/",
+        "cp -a /usr/share/runtime/docs/doc/. /app/share/doc/",
         "cp -a /usr/lib/sdk/vala/bin/. /app/bin/",
         "cp -a /usr/lib/sdk/vala/lib/. /app/lib/",
+        "#cp -a /usr/lib/sdk/vala/share/. /app/share/",
```
[the remark this answers is not in the document]
Comment (lw64) [reply build-aux/re.sonny.Workbench.Devel.json:65]: Probably not right now. But the share/vala-0.x/vapi/ folder is important

[the remark this answers is not in the document]
Comment (lw64) [reply build-aux/re.sonny.Workbench.Devel.json:65]: `org.gnome.Sdk` is supposed to not ship `/usr/share/vala-0.x` anymore. Currently it still does, but it was planned to be already removed.

**Pull request merged**